src/stdlib.js

```diff
@@ -65,39 +79,57 @@ const sharedGlobalPropertyNames = [
   // *** Annex B
 
   'escape',
-  'unescape',
+  'unescape'
 
   // *** ECMA-402
 
-  'Intl'
+  // 'Intl'  // Unstable
 
   // *** ESNext
 
   // 'Realm' // Comes from createRealmGlobalObject()
 ];
 
+const unstableGlobalPropertyNames = [
+  'Date',
+  'Error',
+  'Promise',
+  'Proxy',
+  'RegExp',
+  'Intl'
+];
+
 export function getSharedGlobalDescs(unsafeGlobal) {
-  const descriptors = {
-    // *** 18.1 Value Properties of the Global Object
-    Infinity: { value: Infinity },
-    NaN: { value: NaN },
-    undefined: { value: undefined }
-  };
-
-  for (const name of sharedGlobalPropertyNames) {
-    const desc = getOwnPropertyDescriptor(unsafeGlobal, name);
-    if (desc) {
-      // Abort if an accessor is found on the unsafe global object instead of a
-      // data property. We should never get into this non standard situation.
-      assert('value' in desc, `unexpected accessor on global property: ${name}`);
-
-      descriptors[name] = {
-        value: desc.value,
-        writable: true,
-        configurable: true
-      };
+  const descriptors = {};
+
+  function describe(names, writable, enumerable, configurable) {
+    for (const name of names) {
+      const desc = getOwnPropertyDescriptor(unsafeGlobal, name);
+      if (desc) {
+        // Abort if an accessor is found on the unsafe global object
+        // instead of a data property. We should never get into this
+        // non standard situation.
+        assert('value' in desc,
+               `unexpected accessor on global property: ${name}`);
+
+        descriptors[name] = {
+          value: desc.value,
+          writable,
+          enumerable,
+          configurable
+        };
+      }
     }
   }
 
+  describe(frozenGlobalPropertyNames, false, false, false);
+  // The following is correct but expensive.
+  // describe(stableGlobalPropertyNames, true, false, true);
+  // Instead, for now, we let these get optimized
+  describe(stableGlobalPropertyNames, false, false, false);
```

Comment:
We should provide an option to turn this optimization off, by feeding `true, false, true` here instead. This is low priority until we find a need.